### PR TITLE
[BGP]Adding configuration knob to allow advertise Loopback ipv6 /128 prefix

### PR DESCRIPTION
**Pull request by dgsudharsan**
#### Why I did it
By default when IPv6 address is configured with /128 as subnet mask in Loopback0 interface, it will be advertised as prefix with /64 subnet.
To control this behavior a new field 'bgp_adv_lo_prefix_as_128' is introduced in DEVICE_METADATA table which when set to true will advertise prefix with /128 subnet as it is.

#### How I did it
Modified the jinja templates and yang model to parse this field and advertise route as /128

#### How to verify it
Added UT in sonic-cfggen test as well yang to verify the changes

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Review by volodymyrsamotiy — DISMISSED** [review]

**Comment by StormLiangMS** [comment]
@dgsudharsan  could you elaborate what's use case or issues you want to add this knob? 
from previous code, it intends to use 64 other than 128 for the loopback0.
ipv6 route {{ get_ipv6_loopback_address(LOOPBACK_INTERFACE, "Loopback0") | replace('/128', '/64') | ip_network }}/64 Loopback0

File: dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2 [hunk 1]
```diff
@@ -11,8 +11,12 @@
 ip prefix-list PL_LoopbackV4 permit {{ get_ipv4_loopback_address(LOOPBACK_INTERFACE, "Loopback0") | ip }}/32
 !
 {% if get_ipv6_loopback_address(LOOPBACK_INTERFACE, "Loopback0") != 'None' %}
+{% if DEVICE_METADATA['localhost']['bgp_adv_lo_prefix_as_128'] == 'true' %}
```
[the remark this answers is not in the document]
Comment (dgsudharsan) [reply dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2]: Done

**Comment by dgsudharsan** [comment]
> @dgsudharsan could you elaborate what's use case or issues you want to add this knob? from previous code, it intends to use 64 other than 128 for the loopback0. ipv6 route {{ get_ipv6_loopback_address(LOOPBACK_INTERFACE, "Loopback0") | replace('/128', '/64') | ip_network }}/64 Loopback0

By default SONiC modifies the route advertisement for Loopback network. When Loopback ipv6 address is configured as '/128' it advertises as prefix with '/64'. When two neighboring devices have different '/128' ipv6 address in their loopback but fall within the same /64 subnet, (eg 2001:1::1/128 and 2001:1::2/128) then bgp advertisement will be the same from both routers (2001:1::/64) and it wouldn't be possible to differentiate.

There is no configuration setting in SONiC to overcome this behavior. So introducing a field in device_metadata table which will allow bgp to advertise '/128' Loopback0 ipv6 address as it is.
The reason for not making advertise /128 by default is to avoid having backward compatibility issues with the systems already deployed.

**Pull request merged**